.github/workflows/pr-check.yaml

```diff
@@ -172,7 +172,7 @@ jobs:
 
   lint-format-unit:
     name: linter, formatters and unit tests
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```

Comment:
we should not use latest but use the specific tag (if you want to bump to 22.04)

also, just that that could be a single PR